sdk/dapp-kit/src/hooks/wallet/useAutoConnectionStatus.ts

```diff
@@ -1,12 +1,30 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { useMutationState } from '@tanstack/react-query';
+
+import { walletMutationKeys } from '../../constants/walletMutationKeys.js';
 import { useWalletStore } from './useWalletStore.js';
 
 /**
  * Retrieves the status for the initial wallet auto-connection process.
  */
-export function useAutoConnectionStatus() {
-	// TODO: Replace this with shareable mutation state once we require react-query v5
-	return useWalletStore((state) => state.autoConnectionStatus);
+export function useAutoConnectionStatus(): 'disabled' | 'idle' | 'attempted' {
```

Comment:
can we update the docs for the `settled` -> `attempted` change and also maybe change the description for those states to be clearer (IIRC, I don't think what I originally wrote accurately reflects this behavior)

Reply:
Yeah updated the docs!